core/src/test/scala/unit/kafka/security/auth/ZkAuthorizationTest.scala

```diff
@@ -55,9 +56,11 @@ class ZkAuthorizationTest extends ZooKeeperTestHarness with Logging{
   @Test
   def testIsZkSecurityEnabled() {
     assertTrue(JaasUtils.isZkSecurityEnabled())
+    Configuration.setConfiguration(null)
```

Comment:
@fpj In the test:
 ```
    assertTrue(JaasUtils.isZkSecurityEnabled())
    System.clearProperty(JaasUtils.JAVA_LOGIN_CONFIG_PARAM)
    assertFalse(JaasUtils.isZkSecurityEnabled())
```

With the old code, the first `JaasUtils.isZkSecurityEnabled()` loaded the JAAS configuration. The second checked if the jaas config system property was set, and since it was not set, it returned false. So the second happened to work because of the check for the system property, even though the code was called without resetting the JAAS config, leaving a valid JAAS configuration with ZK logincontext where security was actually enabled for ZK. Similarly for the next check in the same method, the code happened to work because it checked that the file referred to by the system property was readable. Since this PR always tests the current JAAS configuration, any test that expects JAAS configuration to be reloaded, needs to reset the configuration. Hence the additional reset.